src/System.CommandLine/Parsing/SymbolResult.cs

```diff
@@ -44,9 +44,12 @@ Parent switch
         private protected virtual int RemainingArgumentCapacity =>
             MaximumArgumentCapacity() - Tokens.Count;
 
-        internal int MaximumArgumentCapacity() =>
-            Symbol.Arguments()
-                  .Sum(a => a.Arity.MaximumNumberOfValues);
+        internal int MaximumArgumentCapacity()
+        {
+            var maximumArgumentCapacity = Symbol.Arguments()
+                .Sum(a => a.Arity.MaximumNumberOfValues);
+            return (int) Math.Min(maximumArgumentCapacity, int.MaxValue);
```

Comment:
Arguably, this change could be removed; but theoretically someone could go out of their way to cause an overflow.

I'm fine removing this change if you think it's ridiculous to keep it around.